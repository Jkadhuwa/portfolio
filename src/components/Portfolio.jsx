import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import PropTypes from 'prop-types';
import ProjectOverview from '../pages/ProjectsPage';

function Portfolio({ match }) {
  return (
    <Router>
      <Switch>
        <Route exact path={`${match.path}`} component={ProjectOverview} />

        <Route
          exact
          path={`${match.path}/:categoryId`}
          component={ProjectOverview}
        />
      </Switch>
    </Router>
  );
}
Portfolio.propTypes = {
  match: PropTypes.object,
};

export default Portfolio;
