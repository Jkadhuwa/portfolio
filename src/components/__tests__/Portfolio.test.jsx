import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import Portfolio from '../Portfolio';

describe('Test PortfolioButton Component', () => {
  afterEach(cleanup);
  it('Renders  correctly', () => {
    const match = {
      path: '/portfolio',
    };
    const { asFragment } = render(<Portfolio match={match} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
