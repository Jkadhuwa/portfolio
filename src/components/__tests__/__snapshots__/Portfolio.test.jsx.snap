// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PortfolioButton Component Renders  correctly 1`] = `<DocumentFragment />`;
